src/lando/headless_api/tests/test_automation_api.py

```diff
@@ -0,0 +1,678 @@
+    user, token = headless_user
+
+    # Create a job and actions
+    job = AutomationJob.objects.create(status=status)
+    AutomationAction.objects.create(
+        job_id=job, action_type="add-commit", data={"content": "test"}, order=0
+    )
+
+    # Fetch job status.
+    response = client.get(
+        f"/api/job/{job.id}",
+        headers={
+            "User-Agent": "Lando-User/testuser@example.org",
+            "Authorization": f"Bearer {token}",
+        },
+    )
+
+    assert (
+        response.status_code == 200
+    ), "Response code should be 200 when status is retrieved successfully."
+
+    response_data = response.json()
+
+    assert response_data["job_id"] == job.id
+    assert (
+        response_data["message"] == message
+    ), "Response message should align with current job status."
+    # TODO test a few more things? formatting?
+
+
+@pytest.fixture
+def hg_automation_worker(landing_worker_instance):
+    worker = landing_worker_instance(
+        name="automation-worker",
+        scm=SCM_TYPE_HG,
+    )
+    return AutomationWorker(worker)
+
+
+@pytest.fixture
+def git_automation_worker(landing_worker_instance):
+    worker = landing_worker_instance(
+        name="automation-worker",
+        scm=SCM_TYPE_GIT,
+    )
+    return AutomationWorker(worker)
+
+
+@pytest.mark.django_db
+def test_automation_job_add_commit_success_hg(
+    hg_server,
+    hg_clone,
+    hg_automation_worker,
+    repo_mc,
+    monkeypatch,
+    normal_patch,
+    treestatusdouble,
+):
+    repo = repo_mc(SCM_TYPE_HG)
+    scm = repo.scm
+
+    treestatusdouble.open_tree(repo.name)
+
+    # Create a job and actions
+    job = AutomationJob.objects.create(
+        status=JobStatus.SUBMITTED,
+        requester_email="example@example.com",
+        target_repo=repo,
+    )
+    AutomationAction.objects.create(
+        job_id=job,
+        action_type="add-commit",
+        data={
+            "action": "add-commit",
+            "content": normal_patch(1),
+            "patch_format": "hgexport",
+        },
+        order=0,
+    )
+
+    hg_automation_worker.worker_instance.applicable_repos.add(repo)
+
+    # Mock `phab_trigger_repo_update` so we can make sure that it was called.
+    mock_trigger_update = mock.MagicMock()
+    monkeypatch.setattr(
+        "lando.api.legacy.workers.automation_worker.AutomationWorker.phab_trigger_repo_update",
+        mock_trigger_update,
+    )
+
+    scm.push = mock.MagicMock()
+
+    assert hg_automation_worker.run_automation_job(job)
+    assert scm.push.call_count == 1
+    assert len(scm.push.call_args) == 2
+    assert len(scm.push.call_args[0]) == 1
+    assert scm.push.call_args[0][0] == hg_server
+    assert scm.push.call_args[1] == {"push_target": "", "force_push": False}
+    assert job.status == JobStatus.LANDED, job.error
+    assert len(job.landed_commit_id) == 40, "Landed commit ID should be a 40-char SHA."
+
+
+PATCH_GIT_1 = r"""
```

Comment:
Might be worth moving this to conftest along with the HG patches.

Reply:
Will do as a follow-up. :+1: